src/PaymentGateways/Repositories/GatewayPaymentRepository.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Give\PaymentGateways\Repositories;
+
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+use Give_Payment;
+
+/**
+ * This repository return gateway payment data for specific donation.
+ *
+ * @unlreased
+ */
+class GatewayPaymentRepository
+{
+    /**
+     * Get donation title.
+     * This can be use as product item name form gateway payment.
+     *
+     * @param GatewayPaymentData $paymentData
+     * @param string $titleLength
+     *
+     * @return false|int|string
+     * @unlreased
+     */
+    public function getDonationTitle(GatewayPaymentData $paymentData, $titleLength = null)
```

Comment:
This looks similar to my `DonationSummary` class in the new Stripe Credit Card gateway, which might be useful here. I had considered extracting the `DonationSummary` class to the PaymentGateway framework - maybe now is a good time to do that.

Reply:
Otherwise, if this `getDonationTitle` method is specific to PayPal then it should be moved into the specific gateway.

Reply:
@kjohnson Can you extract that `DonationSummary ` class? I already ask you to do that :D

Reply:
@ravinderk ah, yup - Looks like I forgot to do that! Here is a PR that extracts the `DonationSummary` class to the `Framework/PaymentGateways` domain, see https://github.com/impress-org/givewp/pull/6209